app/views/puppetclasses/_class_selection.html.erb

```diff
@@ -45,7 +45,7 @@
     <div class="row">
       <%= render "puppetclasses/classes",
         :puppetclasses => obj.available_puppetclasses,
-        :selected_puppet_classes => obj.classes,
+        :selected_puppet_classes => obj.puppetclasses,
```

Comment:
.classes is important as it includes any classes inherited from a host group (in the case of host#edit) or a parent (in hostgroup#edit)